src/TSTransformer/util/types.ts

```diff
@@ -20,21 +22,85 @@ function getRecursiveBaseTypes(type: ts.InterfaceType) {
 	return result;
 }
 
-function isDefinitelyTypeInner(type: ts.Type, callbacks: Array<TypeCheck>): boolean {
-	if (type.isUnion()) {
-		return type.types.every(t => isDefinitelyTypeInner(t, callbacks));
+function isDefinitelyTypeInner(
```

Comment:
Very unsure about these changes.. I've read through this function a few times and I still find it confusing.

I'm not sure the array form is necessary.
I'd also like to avoid using array.reduce, it makes things unnecessarily complicated

Reply:
I'll see if I can make the reduce call more readable. The array form is discussed in the other review comment.

Reply:
Separated array version. Changed reduce call to a more readable util function.